PRO-16338: Add Husky & Prettier in ProvarDX, Fix linting issues

riteshnahar opened the pull request.

riteshnahar committed:
PRO-16338: Add Husky & Prettier in ProvarDX, Fix linting issues

riteshnahar committed:
PRO-16338: Prettier ignore README.MD & oclif.manifest.json

riteshnahar committed:
PRO-16338: Removed beta paths from README.md

riteshnahar committed:
PRO-16370: Changed project structure from Eclipse project to NPM package structure

riteshnahar committed:
Added LICENCE.md

Added MIT Licence

riteshnahar committed:
Rename LICENSE to LICENSE.md

riteshnahar committed:
PRO-16338: Added copyright notice on all code files

simoncarter-provar commented on README.md:
```diff
@@ -1,8 +1,154 @@
-ProvarDX Salesforce CLI Plugin 
-
-UNDER CONSTRUCTION
+# provartesting/provardx
```
Could we break-up the README and add additional guidelines for CONTRIBUTING, DEVELOPING, CODE_OF_CONDUCT, CHANGELOG so that each file is covering a different area:

- README = usage of the plugin
- CONTRIBUTING = where we work, where we release from, procedures for branching, merging, etc
- CODE_OF_CONDUCT = the expected behaviour of contributors
- DEVELOPING = how to set up the development environment
- CHANGELOG = the changes that have been made.

riteshnahar replied:
Done

simoncarter-provar commented on README.md:
```diff
@@ -1,8 +1,154 @@
 ProvarDX is a Salesforce CLI Plugin for existing Provar customer to allow them to execute Provar Test Cases from the command line and leveraging the Salesforce CLI and SalesforceDX applications. This provides an alternative mechanism to running test cases than running under ANT.
 
-The ProvarDX CLI plugin is free to use (and will be Open Source Licensed) command library that leverages the commercial Provar Runtime library. You must be a Provar customer with a valid paid license to write and maintain your test cases.
+The ProvarDX CLI plugin is a free to use command library that leverages the commercial Provar Runtime library. You must be a Provar customer with a valid paid license to write and maintain your test cases.
```
Can we consolidate these sentences, seems like we're repeating the same message.

riteshnahar replied:
Done

simoncarter-provar commented on README.md:
```diff
@@ -1,8 +1,154 @@
+## Commands
+
+<!-- commands -->
+* [`sfdx provar:compile [-p <string>] [-h <string>] [-c <string>] [--json] [-l <string>]`](#sfdx-provarcompile--p-string--h-string--c-string---json--l-string)
```
Did Richard mention the other day that the /src pageobjects would be compiled automatically when a user runs a test or does this need to be a separate command?

simoncarter-provar commented on README.md:
````diff
@@ -1,8 +1,154 @@
+$ sfdx plugins:install @provartesting/provardx
+```
+
+You can check a successful installation with `sfdx plugins`. Updates are applied when executing `sfdx plugins:update`.
````
To be consistent in how we present commands could you add these into separate code and syntax highlighting blocks under the relevant headers:
- Check the sfdx plugin versions
- Update the sfdx plugin
- Uninstall the sfdx plugin

Worth mentioning that by By default, the salesforce CLI periodically checks for and installs updates for sfdx, should check if this would also occur for its sfdx plugins. To disable sfdx auto-update, set the SFDX_AUTOUPDATE_DISABLE environment variable to true.

simoncarter-provar commented on README.md:
```diff
@@ -1,8 +1,154 @@
+* [`sfdx provar:runtests [-f <string>] [-p <string>] [-o <string>] [-c <string>] [-m <string>] [-s <string>] [--json] [-l <string>]`](#sfdx-provarruntests--f-string--p-string--o-string--c-string--m-string--s-string---json--l-string)
+* [`sfdx provar:validate [-p <string>] [--json] [-l <string>]`](#sfdx-provarvalidate--p-string---json--l-string)
+
+## `sfdx provar:compile [-p <string>] [-h <string>] [-c <string>] [--json] [-l <string>]`
```
Do we need all the compile arguments included in the header?

simoncarter-provar commented on README.md:
```diff
@@ -1,8 +1,154 @@
+* [`sfdx provar:compile [-p <string>] [-h <string>] [-c <string>] [--json] [-l <string>]`](#sfdx-provarcompile--p-string--h-string--c-string---json--l-string)
+* [`sfdx provar:metadatacache [-m <string>] [-c <string>] [-p <string>] [-n <string>] [-o <string>] [--json] [-l <string>]`](#sfdx-provarmetadatacache--m-string--c-string--p-string--n-string--o-string---json--l-string)
+* [`sfdx provar:runtests [-f <string>] [-p <string>] [-o <string>] [-c <string>] [-m <string>] [-s <string>] [--json] [-l <string>]`](#sfdx-provarruntests--f-string--p-string--o-string--c-string--m-string--s-string---json--l-string)
+* [`sfdx provar:validate [-p <string>] [--json] [-l <string>]`](#sfdx-provarvalidate--p-string---json--l-string)
```
So we make this a simple list of commands which link to the sections below as we're duplicating information and the format of the sections below is easier to read

simoncarter-provar commented on README.md:
````diff
@@ -1,8 +1,154 @@
+
+EXAMPLE
+  $ sfdx provar:compile -p './myproperties.json' --json --loglevel SEVERE
+```
````
Does this all need to be in a code block?

riteshnahar replied:
Fixed

simoncarter-provar commented on README.md:
````diff
@@ -1,8 +1,154 @@
+  $ sfdx provar:compile -p './myproperties.json' --json --loglevel SEVERE
+```
+
+_See code: [src/commands/provar/compile.ts](https://github.com/ProvarTesting/provardx/blob/v0.0.1/src/commands/provar/compile.ts)_
````
Where does this link go? this shouldn't be link to a specific version.
This information is for a contributor rather than a user, could we separate this information as described above.

simoncarter-provar commented on .npmignore:
```diff
@@ -0,0 +1,5 @@
+.gitattributes
+.gitignore
+appveyor.yml
+tslint.json
```
should we also ignore the /test directory from the publish npm package?

simoncarter-provar commented on test/commands/hello/org.test.ts:
```diff
@@ -0,0 +1,27 @@
+                'Hello world! This is org: Super Awesome Org and I will be around until Tue Mar 20 2018!'
+            );
+        });
+});
```
Is this providing sufficient test coverage of the source code?

riteshnahar replied:
There is some issue with sfdx plugin and we are not able to run testcases. Once it's resolved will add additional test cases.

simoncarter-provar commented on README.md:
````diff
@@ -1,8 +1,154 @@
+```
+
+_See code: [src/commands/provar/validate.ts](https://github.com/ProvarTesting/provardx/blob/v0.0.1/src/commands/provar/validate.ts)_
+<!-- commandsstop -->
````
Should we reference any documentation or github projects provided by salesforce for sfdx?

simoncarter-provar replied:
Please apply comments from the compile sections to the other sections so the documentation is consistent.

simoncarter-provar commented on test/commands/hello/org.test.ts:
```diff
@@ -0,0 +1,27 @@
+import { expect, test } from '@salesforce/command/lib/test';
+import { ensureJsonMap, ensureString } from '@salesforce/ts-types';
```
Have you run this test?

simoncarter-provar commented on src/commands/provar/runtests.ts:
```diff
@@ -0,0 +1,167 @@
+            const selection = await cli.prompt(
+                'Test plans detected, connection overrides will be ignored, do you wish to continue (Y/N)? '
+            );
+            if (selection === 'n' || selection === 'n') {
```
What's the between of either side of the or operator?

riteshnahar replied:
Done

simoncarter-provar reviewed (changes requested):
I haven't completed the review yet, but here a few comments to get started with.

riteshnahar committed:
PRO-16338: Changed licence from MIT to BSD-3-Clause

riteshnahar committed:
PRO-16338: Incorporated review comments:
- Created CHANGELOG.md, CODE_OF_CONDUCT.md, CONTRIBUTING.md, DEVELOPING.md
- Updated README.md to incorporate comments

simoncarter-provar commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,28 @@
+   starting your work.
+1. Fork this repository and create your _topic_ branch from **development** (see [Branches section](#branches) below).
+1. Edit the code in your fork and add test cases for the same.
+1. If you have added new `sfdx command` then update the documentation (README.md) with the signature of new method.
```
should this be 'provardx command'?

riteshnahar replied:
Done

simoncarter-provar commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,28 @@
+-   Our work happens in _topic_ branches (feature and/or bug-fix).
+    -   feature as well as bug-fix branches are based on `development`
+
+### Merging between branches
```
Please add a section for merging to development into master. We're following Gitflow: topic branch (feature/bug) -> development -> master.

- topic branches: based on development and implement changes for new features or bug fix.
- development: main development branch where all topic branches are merged for ongoing development
- master: production branch of what has been released. Tags will also be created for release versions.

riteshnahar replied:
Added this section.

simoncarter-provar commented on DEVELOPING.md:
```diff
@@ -0,0 +1,76 @@
+4. In the upper left hand corner of VS Code, verify that the "Attach to Remote" launch configuration has been chosen.
+5. Hit the green play button to the left of the "Attach to Remote" launch configuration window. The debugger should now be suspended on the first line of the program.
+6. Hit the green play button at the top middle of VS Code (this play button will be to the right of the play button that you clicked in step #5).
+   <br><img src=".images/vscodeScreenshot.png" width="480" height="278"><br>
```
Do we have this image?

riteshnahar replied:
Added image

simoncarter-provar commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,71 @@
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the Make Positive Provar Ltd. at conduct@provartesting.com. All
```
update to info@provartesting.com

riteshnahar replied:
Done

riteshnahar committed:
PRO-16338: Incorporated review comments

- Changed mail id from conduct@provartesting.com -> info@provartesting.com
- CONTRIBUTING.md added new section for Merging `development` into `master`
- DEVELOPING.md made some edits for documentation

riteshnahar committed:
PRO-16338: Added vscodeScreenshot.png image

simoncarter-provar commented on CONTRIBUTING.md:
```diff
@@ -22,6 +23,14 @@
+
+-   _topic_ branches: based on development and implement changes for new features or bug fix.
+-   `development`: main development branch where all topic branches are merged for ongoing development.
+-   `master`: production branch of what has been released. Tags will also be created for release versions.
```
Sorry, maybe I wasn't clear. Didn't expect you to copy and paste what I wrote, I expected instructions of how we merge changes from the development branch into the master branch when a development cycle finishes.

riteshnahar replied:
Done

riteshnahar committed:
PRO-16338: Incorporated review comment in CONTRIBUTING.md

riteshnahar committed:
PRO-16338: Bumped build version to 0.1.0 and updated CHANGELOG.md
